Exposes the GET /reactions endpoint

chloealee opened the pull request:
#### What's this PR do?
- Updates documentation to include the `GET` `/reaction` endpoint
- Sets up `/GET` `/reaction` endpoint to be able to receive param query filters with the Request (we can add these query params in the future after chatting with data). 
- Moves `total_reactions` out of `meta` data and into reaction body for each reaction in the transformer.

#### How should this be reviewed?
Here is what the return data will look like: 
```
{
  "data": [
    {
      "reaction_id": "1",
      "northstar_id": "Q7COJ2DxqZkT6ZiOVmYyBNa4",
      "postable_id": "0",
      "postable_type": "",
      "created_at": {
        "date": "2016-10-12 17:31:41.000000",
        "timezone_type": 3,
        "timezone": "UTC"
      },
      "updated_at": {
        "date": "2016-10-12 17:31:41.000000",
        "timezone_type": 3,
        "timezone": "UTC"
      },
      "deleted_at": null,
      "total_reactions": null
    },
    {
      "reaction_id": "2",
      "northstar_id": "ILXeYrvXumQ8ld5XAfvyxnWq",
      "postable_id": "0",
      "postable_type": "",
      "created_at": {
        "date": "2016-10-12 17:31:41.000000",
        "timezone_type": 3,
        "timezone": "UTC"
      },
      "updated_at": {
        "date": "2016-10-12 17:31:41.000000",
        "timezone_type": 3,
        "timezone": "UTC"
      },
      "deleted_at": null,
      "total_reactions": null
    },
    {
      "reaction_id": "3",
      "northstar_id": "WqIv9TIAxFbOGT2ZDPb0loLe",
      "postable_id": "0",
      "postable_type": "",
      "created_at": {
        "date": "2016-10-12 17:31:41.000000",
        "timezone_type": 3,
        "timezone": "UTC"
      },
      "updated_at": {
        "date": "2016-10-12 17:31:41.000000",
        "timezone_type": 3,
        "timezone": "UTC"
      },
      "deleted_at": null,
      "total_reactions": null
    },
  ],
  "meta": {
    "pagination": {
      "total": 3,
      "count": 3,
      "per_page": 20,
      "current_page": 1,
      "total_pages": 1,
      "links": []
    }
  }
}
```
Should anything be changed? 

#### Any background context you want to provide?
This will only return reactions that are currently live (will not return soft deleted reactions). 

#### Relevant tickets
Fixes #143 

#### Checklist
- [ ] Documentation added for new features/changed endpoints.
- [ ] Tested on staging.
- [ ] Pinged a PM if this is a larger PR that would benefit from some additional testing love.

chloealee commented on app/Http/Controllers/Api/ReactionController.php:
```diff
@@ -24,6 +25,22 @@ public function __construct()
     }
```
@sbsmith86 @katiecrane do we want this endpoint to be available to unauthorized users? If so, we might want to add middleware back into the `routes` file and maybe make a separate section for all `GET` endpoints (like `GET` `/activity`)?

sbsmith86 replied:
Hmmm do we currently expose this to unauthenticated user?

chloealee replied:
we do for the `GET /activity` endpoint, which was requested from data!

katiecrane commented on documentation/endpoints/reactions.md:
```diff
@@ -1,5 +1,84 @@
+      "postable_id": "4",
+      "postable_type": "Rogue\\Models\\Photo",
+      "created_at": {
+        "date": "2016-10-12 17:31:41.000000",
```
@chloealee I think our other timestamps are formatted like this but without the `.000000`, do you know why that appears here?

chloealee replied:
@katiecrane ah thank you for catching that! I forgot to format with `toIso8601String()` in the transformer - updated!

katiecrane replied:
@chloealee Turns out I didn't know what format we used elsewhere! You switched it to what we actually do use elsewhere though awesome!

sbsmith86 commented:
@chloealeewait sorry is this for `/reactions` or more work on `/activities`

sbsmith86 commented:
i think i'm confused about this 

> Moves total_reactions out of meta data and into reaction body for each reaction in the transformer.

chloealee commented:
@sbsmith86 this is to expose the `GET /reactions` endpoint! 

RE: 

> Moves total_reactions out of meta data and into reaction body for each reaction in the transformer.

When I worked on the `POST /reactions` endpoint, I added `total_reactions` into the `meta` data of the transformer. What was returned when hitting the `POST /reactions` endpoint looked like this: 
```
{
  "data": {
    "reaction_id": "22",
    "northstar_id": "8888",
    "postable_id": "352",
    "postable_type": "Rogue\\Models\\Photo",
    "created_at": {
      "date": "2016-10-12T17:31:41+0000",
      "timezone_type": 3,
      "timezone": "UTC"
    },
    "updated_at": {
      "date": "2016-10-12T17:31:41+0000",
      "timezone_type": 3,
      "timezone": "UTC"
    },
    "deleted_at": {
      "date": "2016-10-12T17:31:41+0000",
      "timezone_type": 3,
      "timezone": "UTC"
    },
  },
  "meta": {
    "action": "unliked",
    "total_reactions": 1
  }
}
```

In order to have that data returned for reach reaction in the `GET /reaction` endpoint, I have moved it into the Reaction Transformer and out of the `meta` data. Now, the `POST /reaction` response looks like this: 
```
{
  "data": {
    "reaction_id": "22",
    "northstar_id": "8888",
    "postable_id": "352",
    "postable_type": "Rogue\\Models\\Photo",
    "created_at": {
      "date": "2016-10-12T17:31:41+0000",
      "timezone_type": 3,
      "timezone": "UTC"
    },
    "updated_at": {
      "date": "2016-10-12T17:31:41+0000",
      "timezone_type": 3,
      "timezone": "UTC"
    },
    "deleted_at": {
      "date": "2016-10-12T17:31:41+0000",
      "timezone_type": 3,
      "timezone": "UTC"
    },
    "total_reactions": 1
  },
  "meta": {
    "action": "unliked"
  }
}
```

sbsmith86 commented:
@chloealee Sorry, I think im confused about this endpoint. It seems like the `/activity` endpoint would suffice to get a count of how many reactions something got. So like how would this be used differently? Reaction ids, don't seem that useful outside of rogue.

chloealee commented:
hmm @sbsmith86 yeah I think you're right, I didn't think about this enough. I think the idea was to have a `GET /reactions` endpoint so we could expose the total amount of reactions a post got in the gallery but once Rogue powers the gallery, we shouldn't need this endpoint. @ngjo am I missing anything else for why we wanted this?

sbsmith86 commented:
@chloealee yea ideally, we would just be using the `/activities` endpoint to power the gallery. I don't see a reason we would need to query for reactions separately.

chloealee commented:
that makes sense to me @sbsmith86 ! So I guess then we'd need to make cards to: 
- create a new event when a reaction is created (and logic/associations that goes along with that)
- add query param to filter activity to return reaction data? 
cc @ngjo

sbsmith86 commented:
@chloealee we definitely want the reaction event logic but i don't think we need the query param right now. 

In my head, phoenix will query for a bunch of reportbacks and we return each reportback with the reaction count for each reportback in the response.

sbsmith86 commented:
Like in the activity response outlined here https://github.com/DoSomething/rogue/blob/master/documentation/endpoints/activity.md

Maybe we could embed the total reaction count in each post object. So it's like

```
          {
            "postable_id": 74,
            "post_event_id": 124,
            "submission_type": "user",
            "postable_type": "Rogue\\Models\\Photo",
            "total_reactions": 21,
            "content": {
              "media": {
                "url": "https://s3.amazonaws.com/ds-rogue-prod/uploads/reportback-items/15-1484944153.jpeg",
                "edited_url": "https://s3.amazonaws.com/ds-rogue-prod/uploads/reportback-items/edited_15-1484944153.jpeg"
              },
              "caption": "lolz",
              "status": "pending",
              "remote_addr": "10.0.2.2",
              "post_source": "phoenix-web",
              "created_at": "2017-01-20T20:29:14+0000",
              "updated_at": "2017-01-20T20:29:14+0000",
            }
          }
```

chloealee commented:
@sbsmith86 ah I love that idea! Before closing and making those issues, just want to double check with @ngjo I'm not missing anything for the purpose for exposing the `GET /reactions` endpoint.

ngjo commented:
@chloealee sorry for the delay! i've been in meetings all day - and no, I don't think there was any other reason for exposing that endpoint. if it all works with the /activity endpoint that's great!

chloealee commented:
no worries, I figured @ngjo ! Thanks for confirming!
